src/npctalk.cpp

```diff
@@ -2073,6 +2204,10 @@ void parse_tags( std::string &phrase, const talker &u, const talker &me, const d
             return;
         }
 
+        const auto recursive = [&u, &me, &d, &item_type]( std::string & new_phrase ) {
+            parse_tags( new_phrase, u, me, d, item_type );
+        };
+
```

Comment:
Perhaps instead of this, a callback can be added to the parameters of `snippet_library::expand` to handle any unhandled tags, so we don't have to handle these tags with separate code.

Reply:
`replace_city_tag` and `replace_keybind_tag` from `output.cpp` can also be handled by the snippet library, although we need to handle the conflict between the `<city>` tags in `npctalk.cpp` and `output.cpp`.